internal/controllers/topology/cluster/structuredmerge/dryrun.go

```diff
@@ -147,6 +165,11 @@ func dryRunSSAPatch(ctx context.Context, dryRunCtx *dryRunSSAPatchInput) (bool,
 	hasChanges := len(diff.Object) > 0
 	_, hasSpecChanges := diff.Object["spec"]
 
+	// If there were no changes add the request identifier to the tracker.
+	if !hasChanges {
+		dryRunCtx.ssaRequestTracker.Add(requestIdentifier)
+	}
+
```

Comment:
if we can refactor ssa.Run as suggested below, I'm wondering if we can provide a DryRun option and use a single mechanism to add things to the tracker